File: backend/KtorApp/src/main/kotlin/com/example/models/TextBlock.kt
package com.example.models

import kotlinx.serialization.Serializable
import org.bson.Document
import org.bson.types.ObjectId

@Serializable
data class TextBlock(
    val _id: String, // Используем _id вместо idBlock
    val original: String,
    val numberChapter: Int = 0,
    val summary: String? = null,
    val summaryTime: String? = null,
    val questions: List<String> = emptyList(),
    val rightAnswers: List<String> = emptyList()
) {
    fun toDocument(): Document {
        return Document("_id", ObjectId(_id)) // Используем _id вместо idBlock
            .append("original", original)
            .append("numberChapter", numberChapter)
            .append("summary", summary)
            .append("summaryTime", summaryTime)
            .append("questions", questions)
            .append("rightAnswers", rightAnswers)
    }
}

File: backend/models/TextBlock.py
from typing import Optional, List
from pydantic import BaseModel
from bson import ObjectId

class TextBlock(BaseModel):
    _id: str
    original: str
    numberChapter: int = 0
    summary: Optional[str] = None
    summaryTime: Optional[str] = None
    questions: List[str] = []
    rightAnswers: List[str] = []

    def to_document(self) -> dict:
        return {
            "_id": ObjectId(self._id),
            "original": self.original,
            "numberChapter": self.numberChapter,
            "summary": self.summary,
            "summaryTime": self.summaryTime,
            "questions": self.questions,
            "rightAnswers": self.rightAnswers
        }
